lib/src/Uncertainty/Algorithm/Simulation/openturns/NAIS.hxx

```diff
@@ -0,0 +1,105 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief NAIS implement Non Parametric Adaptive Importance Sampling algorithm
+ *
+ *  Copyright 2005-2020 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_NAIS_HXX
+#define OPENTURNS_NAIS_HXX
+
+#include "openturns/PersistentObject.hxx"
+#include "openturns/StorageManager.hxx"
+#include "openturns/Point.hxx"
+#include "openturns/ThresholdEvent.hxx"
+#include "openturns/EventSimulation.hxx"
+#include "openturns/Mixture.hxx"
+#include "openturns/ComposedDistribution.hxx"
+#include "openturns/Normal.hxx"
```

Comment:
you probably can move Normal, ThresholdEvent, EventSimulation, Mixture, ComposedDistribution headers include into the .cxx because they're only needed there, PersistentObject & Point should be useless too